### ISSUE TITLE
uh-oh: kevent() failed BADF

### ISSUE THREAD

**Issue by vjpr** [oven-sh/bun#3439]
### What version of Bun is running?

0.6.11 canary 20230628Wed

### What platform is your computer?

Darwin 22.5.0 arm64 arm

### What steps can reproduce the bug?

Happened just after `Bun.build(...)`...not the first time it was run though...

```
uh-oh: kevent() failed BADF
bun will crash now 😭😭😭

----- bun meta -----
Bun v0.6.11 (2eb885c1) macOS Silicon 22.5.0
AutoCommand: define
Elapsed: 89389ms | User: 1054ms | Sys: 2749ms
RSS: 0.39GB | Peak: 0.39GB | Commit: 0.30GB | Faults: 499
----- bun meta -----

0   0x102c24ebc WTFGetBacktrace
1   ??? Bun__crashReportDumpStackTrace

Crash report saved to:
  ~/.bun/.bun-crash/v0.6.11-1687938191893.crash

Search GitHub issues https://bun.sh/issues or ask for #help in https://bun.sh/discord

thread 23302833 panic: kevent() failed BADF
```

**Comment by simylein** [comment]
I cannot manage to reproduce that, is this still happening with bun version 0.7.1?
Are you just running `Bun.build()` or what would be the minimal reproduction?

**Comment by vjpr** [comment]
I will try to put together a reproduction. Could have been related to this one that was happening at the same time: https://github.com/oven-sh/bun/issues/3364

**Comment by Jarred-Sumner** [comment]
I believe this was fixed in Bun v1.1.8 or so, there was a race condition where multiple of the same thread could be started at once
